app/src/main/java/org/listenbrainz/android/ui/components/LBLoadingAnimation.kt

```diff
@@ -0,0 +1,127 @@
+package org.listenbrainz.android.ui.components
+
+import androidx.compose.animation.core.Animatable
+import androidx.compose.animation.core.animateDpAsState
+import androidx.compose.animation.core.animateFloatAsState
+import androidx.compose.animation.core.tween
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.drawWithCache
+import androidx.compose.ui.geometry.Offset
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.Path
+import androidx.compose.ui.graphics.drawscope.rotate
+import androidx.compose.ui.graphics.drawscope.translate
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.Dp
+import androidx.compose.ui.unit.dp
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+/** ListenBrainz Logo loading animation.
+ *
+ * To fit the animation, actual free space required is ( 3*[logoSize] x [logoSize] ).
+ * @param logoSize Size of the logo represented as Size([logoSize],[logoSize]).
+ * @param middleGap Gap between the parallelograms. Automatically adjusted with [logoSize].
+ * @param sideOffset Half the total length of the outer side of the parallelogram. Automatically adjusted with [logoSize].
+ * @param backgroundColor Background of the animation. Defaults to [Color.Transparent].
+ * @param lb_orange Tint of orange rectangle of the logo.
+ * @param lb_purple Tint of purple rectangle of the logo.
+ * @param translateDuration Duration of translation and alpha animation.
+ * @param rotateDuration Duration of rotate animation.
+ */
+@Preview
+@Composable
+fun LBLoadingAnimation(
+    logoSize: Dp = 60.dp,
+    middleGap: Dp = logoSize / 20,
+    sideOffset: Dp = logoSize / 4,
+    backgroundColor: Color = Color.Transparent,
+    lb_orange: Color = Color(0xFFEA743B),
```

Comment:
We should specify generic names for these parameters. Also, this composable could just be a LoadingAnimation.

Reply:
These are actually supposed to be constant as is and these are the best generic names my brain could cook up 😭

Reply:
leftColor and rightColor could be the names @07jasjeet . The idea is to generalize components with dynamic animation options/colors, etc.

Reply:
oh the color params I see.

Reply:
Done